doc/source/starting-ray.rst

```diff
@@ -120,79 +104,67 @@ To stop or restart Ray, use the shutdown API.
 
-          To connect your program to the Ray cluster, run it like this:
+    You can connect to this Ray runtime by starting a Python process that calls the following on the same node as where you ran ``ray start``:
 
-              .. code-block:: bash
+    .. code-block:: python
 
-                  java -classpath /path/to/jars/ \
-                    -Dray.address=<address> \
-                    <classname> <args>
+      # This must
+      import ray
+      ray.init(address='auto')
 
-          .. note:: Specifying ``auto`` as the address hasn't been implemented in Java yet. You need to provide the actual address. You can find the address of the server from the output of the ``ray up`` command.
+  .. group-tab:: java
 
-      Your driver code **only** needs to execute on one machine in the cluster (usually the head node).
 
-      .. note:: Without the address parameter, your Ray program will only be parallelized across a single machine!
+    If you want to run Java code, you need to specify the classpath via the ``--code-search-path`` option. See :ref:`code_search_path` for more details.
 
-Manual cluster setup
-~~~~~~~~~~~~~~~~~~~~
+    .. code-block:: bash
 
-You can also use the manual cluster setup (:ref:`ref-cluster-setup`) by running initialization commands on each node.
+      $ ray start ... --code-search-path=/path/to/jars
 
-**On the head node**:
 
-.. code-block:: bash
+You can connect other nodes to the head node, creating a Ray cluster by also calling ``ray start`` on those nodes. See :ref:`manual-cluster` for more details. Calling ``ray.init(address="auto")`` on any of the cluster machines will connect to the ray cluster.
 
-    # If the ``--redis-port`` argument is omitted, Ray will choose a port at random.
-    $ ray start --head --redis-port=6379
+.. _start-ray-up:
 
-The command will print out the address of the Redis server that was started (and some other address information).
+Launching a Ray cluster (``ray up``)
+------------------------------------
 
-**Then on all of the other nodes**, run the following. Make sure to replace ``<address>`` with the value printed by the command on the head node (it should look something like ``123.45.67.89:6379``).
+Ray clusters can be launched with the :ref:`Cluster Launcher <ref-automatic-cluster>`.
+The ``ray up`` command uses the Ray cluster launcher to start a cluster on the cloud, creating a designated "head node" and worker nodes. Underneath the hood, it automatically calls ``ray start`` to create a Ray cluster.
 
-.. code-block:: bash
+Your code **only** needs to execute on one machine in the cluster (usually the head node). Read more about :ref:`running programs on a Ray cluster <using-ray-on-a-cluster>`.
 
-    $ ray start --address=<address>
+To connect to the existing cluster, similar to the method outlined in :ref:`start-ray-cli`, you must call ``ray.init`` and specify the address of the Ray cluster when initializing Ray in your code. This causes Ray to .
```

Comment:
There seem to be missing a half sentence at the end here

Reply:
```suggestion
To connect to the existing cluster, similar to the method outlined in :ref:`start-ray-cli`, you must call ``ray.init`` and specify the address of the Ray cluster when initializing Ray in your code. This allows Ray to connect to the cluster.
```